gen/ad_group_vsup_o365

```diff
@@ -0,0 +1,183 @@
+		my $expirationManual = $data->getUserAttributeValue(member => $memberId, attrName => $A_EXPIRATION_MANUAL);
+
+		my $status = getStatus($expirationKOS, $expirationDC2, $expirationManual);
+
+		# skip "vsup expired" members
+		next unless ($status eq "active");
+
+		my $serviceUser = $data->getUserAttributeValue( member => $memberId, attrName => $A_IS_SERVICE );
+
+		if (defined $serviceUser and ($serviceUser == 1)) {
+			# store which service users (their DN) are on this resource
+			$usersByResource->{$group}->{"CN=" . $login . "," . $baseServiceDN} = 1
+		} else {
+			# store which normal users (their DN) are on this resource
+			$usersByResource->{$group}->{"CN=" . $login . "," . $baseDN} = 1
+		}
+
+
+	}
+
+}
+
+#
+# Print group data LDIF
+#
+open FILE,">:encoding(UTF-8)","$fileName" or die "Cannot open $fileName: $! \n";
+
+for my $group (sort keys %$groups) {
+
+	print FILE "dn: CN=" . $group . "," . $baseGroupDN . "\n";
+	print FILE "cn: " . $group . "\n";
+	print FILE "objectClass: group\n";
+	print FILE "objectClass: top\n";
+
+	my @groupMembers = sort keys %{$usersByResource->{$group}};
+	for my $member (@groupMembers) {
+		print FILE "member: " . $member . "\n";
+	}
+
+	# there must be empty line after each entry
+	print FILE "\n";
+
+}
+
+close FILE;
+
+perunServicesInit::finalize;
+
+
+#
+# active - výchozí stav, netřeba nastavovat při vytváření schránky, pouze pokud byl účet předtím v jiném stavu
+# locked - uzamčen pro přihlašování, maily jsou doručovány
+# closed - nelze se přihlásit, maily nejsou doručovány - soft-delete
+#
+sub getStatus() {
+
+	# read input
+	my $expirationKos = shift;
+	my $expirationDc2 = shift;
+	my $expirationMan = shift;
+	# parse to time or undef
+	my $expirationKosTime = ($expirationKos) ? Time::Piece->strptime($expirationKos,"%Y-%m-%d") : undef;
+	my $expirationDc2Time = ($expirationDc2) ? Time::Piece->strptime($expirationDc2,"%Y-%m-%d") : undef;
+	my $expirationManTime = ($expirationMan) ? Time::Piece->strptime($expirationMan,"%Y-%m-%d") : undef;
+
+	my @expirations = ();
+	if (defined $expirationKosTime) { push(@expirations, $expirationKosTime->epoch); }
+	if (defined $expirationDc2Time) { push(@expirations, $expirationDc2Time->epoch); }
+	if (defined $expirationManTime) { push(@expirations, $expirationManTime->epoch); }
+
+	# sort all expirations
+	my @sorted_expirations = sort { $a <=> $b } @expirations;
+
+	my $latest_expiration = $sorted_expirations[$#sorted_expirations];
+	my $currentDate = Time::Piece->strptime(localtime->ymd,"%Y-%m-%d");
+
+	if (!defined $expirationKos and !defined $expirationDc2 and !defined $expirationMan) {
+		# if no expiration set in source data - take as "never"
+		return 'active';
+	}
+
+	# Add time 23:59:59 to the date, since we want accounts to be active on the last day
+	$latest_expiration = $latest_expiration + 86399;
+
+	if ($latest_expiration > $currentDate->epoch) {
+		return 'active';
+	}
+
+	# (will) expire by studies - add 14 days grace period
```

Comment:
This service should have only +14 days grace period, so disregarding users relation (employee/student) - meaning expiration source - you can always check only latest expiration + 14 days against the current date. Also then you can probably return simple 0/1 from this method.